source/portable/os/ota_os_freertos.c

```diff
@@ -43,7 +43,7 @@
 #define MAX_MSG_SIZE    sizeof( OtaEventMsg_t )
 
 /* Array containing pointer to the OTA event structures used to send events to the OTA task. */
-static OtaEventMsg_t queueData[ MAX_MESSAGES ];
+static OtaEventMsg_t queueData[ MAX_MESSAGES * MAX_MSG_SIZE ];
```

Comment:
This array does now allocate `MAX_MESSAGES * MAX_MSG_SIZE * sizeof(OtaEventMsg_t)` bytes, which is slightly too much ( by about a factor of `MAX_MSG_SIZE`).